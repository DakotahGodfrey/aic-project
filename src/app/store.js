import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
