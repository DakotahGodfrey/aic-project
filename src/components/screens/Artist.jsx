import {
  fetchArtistData,
  getMoreArtworksByArtist,
  selectArtist,
} from 'app/slices/artistSlice';
import SectionHeader from 'components/layout/headers/SectionHeader';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { generatePlaceholderArray } from 'utils/helpers';
import Bio from 'components/layout/sections/Bio';
import Feed from 'components/layout/Feed';
import ReactHtmlParser from 'react-html-parser';
import FeaturedArtworks from 'components/layout/sections/FeaturedArtworks';
import useViewport from 'utils/hooks/useViewport';
import { mediaQueries } from 'utils/data';
import LoadMoreButton from 'components/layout/buttons/LoadMoreButton';
import BigImageHero from 'components/layout/headers/BigImageHero';

const Artist = ({ match }) => {
  const dispatch = useDispatch();
  const artist = useSelector(selectArtist);
  const { artistData, artistArtworks, status, currentPage } = artist;
  document.title = `AIC Explorer | ${artistData.title}`;

  React.useEffect(() => {
    dispatch(fetchArtistData(match.params.id));
  }, [dispatch, match.params.id]);

  const handleLoadMoreClick = () => {
    const nextPage = currentPage + 1;
    console.log(nextPage);
    dispatch(
      getMoreArtworksByArtist({ artist: artistData.title, page: nextPage })
    );
  };

  const { width } = useViewport();
  const { small, large } = mediaQueries;
  const isSmall = width < small;
  const isNarrow = width < large;
  return (
    <>
      <BigImageHero
        image={artistArtworks[0] ?? generatePlaceholderArray(1)[0]}
      />
      <Feed>
        {artistData.intro_copy && (
          <>
            <SectionHeader
              css={`
                padding: 0;
              `}
              children={
                <h2>About {artistData.title ? artistData.title : ''}</h2>
              }
            />{' '}
            <Bio>{ReactHtmlParser(artistData.intro_copy)}</Bio>
          </>
        )}
        <SectionHeader children={<h3>Related Artworks</h3>} />
        {status === 'resolved' ? (
          <FeaturedArtworks
            artworks={artistArtworks}
            columns={isSmall ? 1 : isNarrow ? 2 : 3}
          />
        ) : (
          <FeaturedArtworks
            artworks={generatePlaceholderArray(6)}
            columns={isSmall ? 1 : isNarrow ? 2 : 3}
          />
        )}
        <LoadMoreButton
          onClick={() => {
            handleLoadMoreClick();
          }}
          disabled={status === 'pending'}
          children={
            <>
              Load More <i className='material-icons'>keyboard_arrow_down</i>
            </>
          }
        />
      </Feed>
    </>
  );
};

export default Artist;
