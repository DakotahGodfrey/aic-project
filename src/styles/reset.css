*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
