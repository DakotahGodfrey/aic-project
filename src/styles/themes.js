const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
