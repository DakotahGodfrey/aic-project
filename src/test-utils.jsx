import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
