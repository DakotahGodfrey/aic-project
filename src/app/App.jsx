import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
