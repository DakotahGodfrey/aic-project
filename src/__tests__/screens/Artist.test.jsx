import { render, screen } from 'test-utils';
import Artist from 'components/screens/Artist';
import { defaultTheme } from 'styles/themes';
import { store } from 'app/store';

let match = {
  params: {},
};
match.params.id = 1;
describe('Artist', () => {
  it('should render', () => {
    const { container } = render(<Artist match={match} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(container).toMatchSnapshot();
  });
  it('renders a header with the correct text', () => {
    const { getByRole } = render(<Artist match={match} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(getByRole('banner')).toHaveTextContent(/related artworks/i);
  });
});
