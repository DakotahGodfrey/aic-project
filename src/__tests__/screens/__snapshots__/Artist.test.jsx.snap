// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist should render 1`] = `
<div>
  <main
    class="sc-gtsrHT cYYOiy"
  >
    <header
      class="SectionHeader-sc-1gpo88d-0 ZdQJP"
    >
      <h3>
        Related Artworks
      </h3>
    </header>
    <section
      class="sc-dlnjwi kHKgNa"
    >
      <figure
        class="sc-hKFxyN gFUwJC"
      >
        <img
          src="placeholder.svg"
        />
        <figcaption>
          <div
            class="sc-iCoGMd gMBxgk"
          >
            Loading
            <div>
              .
            </div>
            <div>
              .
            </div>
            <div>
              .
            </div>
          </div>
        </figcaption>
      </figure>
      <figure
        class="sc-hKFxyN gFUwJC"
      >
        <img
          src="placeholder.svg"
        />
        <figcaption>
          <div
            class="sc-iCoGMd gMBxgk"
          >
            Loading
            <div>
              .
            </div>
            <div>
              .
            </div>
            <div>
              .
            </div>
          </div>
        </figcaption>
      </figure>
      <figure
        class="sc-hKFxyN gFUwJC"
      >
        <img
          src="placeholder.svg"
        />
        <figcaption>
          <div
            class="sc-iCoGMd gMBxgk"
          >
            Loading
            <div>
              .
            </div>
            <div>
              .
            </div>
            <div>
              .
            </div>
          </div>
        </figcaption>
      </figure>
      <figure
        class="sc-hKFxyN gFUwJC"
      >
        <img
          src="placeholder.svg"
        />
        <figcaption>
          <div
            class="sc-iCoGMd gMBxgk"
          >
            Loading
            <div>
              .
            </div>
            <div>
              .
            </div>
            <div>
              .
            </div>
          </div>
        </figcaption>
      </figure>
      <figure
        class="sc-hKFxyN gFUwJC"
      >
        <img
          src="placeholder.svg"
        />
        <figcaption>
          <div
            class="sc-iCoGMd gMBxgk"
          >
            Loading
            <div>
              .
            </div>
            <div>
              .
            </div>
            <div>
              .
            </div>
          </div>
        </figcaption>
      </figure>
      <figure
        class="sc-hKFxyN gFUwJC"
      >
        <img
          src="placeholder.svg"
        />
        <figcaption>
          <div
            class="sc-iCoGMd gMBxgk"
          >
            Loading
            <div>
              .
            </div>
            <div>
              .
            </div>
            <div>
              .
            </div>
          </div>
        </figcaption>
      </figure>
    </section>
    <button
      aria-label="Load More"
      class="sc-fujyAs hcZibb"
      disabled=""
    >
      Load More
       
      <i
        aria-hidden="true"
        class="material-icons"
      >
        keyboard_arrow_down
      </i>
    </button>
  </main>
</div>
`;
